matlab/src/cpp/arrow/matlab/c/proxy/array.cc

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstddef>
+#include "arrow/c/abi.h"
+
+#include "arrow/matlab/c/proxy/array.h"
+
+#include "libmexclass/proxy/Proxy.h"
+
+namespace arrow::matlab::c::proxy {
+
+Array::Array() : arrowArray{(struct ArrowArray*)malloc(sizeof(struct ArrowArray))} {
+  REGISTER_METHOD(Array, getAddress);
+}
+
+Array::~Array() {
+  if (arrowArray != NULL) {
+    if (arrowArray->release != NULL) {
+      arrowArray->release(arrowArray);
+      arrowArray->release = NULL;
+    }
+    //free(arrowArray);
```

Comment:
Why do we need to comment out this?
We must call `free()` for `malloc()`-ed memory.

Reply:
Oh sorry if why I did this wasn't clear. I was trying to determine if this `free` call was responsible for the crashes I've been seeing in the CI checks.

Reply:
Changed `arrowArray` to be a `struct ArrowArray` instead of a `struct ArrowArray*`.